File: dp_labs_and_solutions/dp_labs_and_solutions/dp_matlab_labs_and_solutions/adapter/solution/Table.m
% Table.m

classdef (Abstract) Table < handle
	methods (Abstract)
		result = getNumberOfRows(theTable)
		result = getNumberOfColumns(theTable)
		result = getColumnName(theTable, aColumn)
		result = getValue(theTable, aRow, aColumn)
	end

	methods (Sealed)
		function display(theTable)
			for aColumn = 1:theTable.getNumberOfColumns
				fprintf('| %s ', theTable.getColumnName(aColumn));
			end
			fprintf('|\n');

			for aRow = 1:theTable.getNumberOfRows
				for aColumn = 1:theTable.getNumberOfColumns
					fprintf('| %s ', theTable.getValue(aRow, aColumn));
				end
				fprintf('|\n');
			end
		end
	end
end

File: dp_labs_and_solutions/original/dp_labs_and_solutions/dp_java_labs_and_solutions/adapter/solution/Table.java
// Table.java

public abstract class Table
{
	public abstract int getNumberOfRows();
	public abstract int getNumberOfColumns();
	public abstract String getColumnName(int aColumn);
	public abstract Object getValue(int aRow, int aColumn);

	public final void display()
	{
		for (int aColumn = 0; aColumn < getNumberOfColumns(); ++aColumn)
		{
			System.out.print("| " + getColumnName(aColumn) + " ");
		}
		System.out.println("|");

		for (int aRow = 0; aRow < getNumberOfRows(); ++aRow)
		{
			for (int aColumn = 0; aColumn < getNumberOfColumns(); ++aColumn)
			{
				System.out.print("| " + getValue(aRow, aColumn) + " ");
			}
			System.out.println("|");
		}
	}
}
